src/modules/dashboard/components/Whitelist/CSVUploader/CSVUploader.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react';
+import Papa, { ParseResult } from 'papaparse';
+import { useField } from 'formik';
+import { MessageDescriptor } from 'react-intl';
+import isNil from 'lodash/isNil';
+
+import FileUpload, { UploadFile } from '~core/FileUpload';
+import { isAddress } from '~utils/web3';
+
+import CSVUploaderItem from './CSVUploaderItem';
+
+interface Props {
+  name: string;
+  error?: string | MessageDescriptor;
+  processingData: boolean;
+  setProcessingData: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const MIME_TYPES = ['text/csv'];
+
+const CSVUploader = ({
+  name,
+  error,
+  processingData,
+  setProcessingData,
+}: Props) => {
+  const [CSVFile, setCSVFile] = useState<File | null>(null);
+  const [parsedCSV, setParsedCSV] = useState<ParseResult<unknown> | null>(null);
+  const [, { value }, { setValue }] = useField<UploadFile[]>(name);
+
+  const handleUploadError = async () => {
+    if (isNil(value[0].parsedData)) {
+      await setValue([{ ...value[0], parsedData: [] }]);
+    }
+  };
+
+  useEffect(() => {
+    if (CSVFile && !parsedCSV) {
+      Papa.parse(CSVFile, {
+        complete: setParsedCSV,
+        header: true,
+      });
+    } else if (!CSVFile && parsedCSV) {
+      setParsedCSV(null);
+    }
+  }, [CSVFile, parsedCSV]);
+
+  useEffect(() => {
+    if (parsedCSV && value[0] && isNil(value[0].parsedData)) {
+      const headerExpectedName = 'Whitelist Address';
```

Comment:
We don't really expect this exact header name to be in the file, we just expect **a header** to be available on the first line

Reply:
I'm gonna make it dynamic then :ok_hand: